File: src/main/java/com/scottscmo/song/Song.java
package com.scottscmo.song;

import java.util.List;
import java.util.Map;

public record Song(
        String title,
        String collection,
        String description,
        List<Verse> lyrics,
        List<String> verseOrder) {

    public Map<String, String[]> getVerseText(String verseNum) {
        Verse verse = this.lyrics.stream()
                .filter(v -> v.verse().equals(verseNum))
                .findFirst().get();
        if (verse != null) {
            return verse.text();
        }
        return null;
    }
}

File: src/main/kotlin/com/scottscmo/song/Song.kt
package com.scottscmo.song

class Song() {
    lateinit var title: String
    lateinit var description: String
    lateinit var collection: String
    lateinit var lyrics: List<Verse>
    lateinit var verseOrder: List<String>

    fun getVerseText(verseNum: String): Map<String, List<String>>? {
        val verse = this.lyrics.firstOrNull { it.verse == verseNum }
        return verse?.textLines()
    }
}
